src/runtime_src/core/common/system.h

```diff
@@ -67,6 +67,11 @@ class system
   virtual device::id_type
   get_device_id(const std::string& str) const;
 
+  /**
+   */
+  virtual std::tuple<uint16_t, uint16_t, uint16_t, uint16_t>
+  get_bdf_info(device::id_type id, bool is_user = true) const = 0;//{ return std::make_tuple(0, 0, 0, 0); }
```

Comment:
Let's remove the trailing comment and instead expand the leading comment.
I am thinking maybe you do want the default implementation as indicated in the trailing comment and not have to change the emulation shims that do not support bdf.

Reply:
I was not sure what was the best choice as total devices is implemented everywhere but I am glad that is the right way to do it!